pkg/volume/flocker/flocker_volume.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flocker
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+type volumeManager interface {
+	// Creates a volume
+	CreateVolume(provisioner *flockerVolumeProvisioner) (datasetUUID string, volumeSizeGB int, labels map[string]string, err error)
+	// Deletes a volume
+	DeleteVolume(deleter *flockerVolumeDeleter) error
+}
+
+type flockerVolumeDeleter struct {
+	*flockerVolume
+}
+
+var _ volume.Deleter = &flockerVolumeDeleter{}
+
+func (b *flockerVolumeDeleter) GetPath() string {
+	return getPath(b.podUID, b.volName, b.plugin.host)
+}
+
+func (d *flockerVolumeDeleter) Delete() error {
+	return d.manager.DeleteVolume(d)
+}
+
+type flockerVolumeProvisioner struct {
+	*flockerVolume
+	options volume.VolumeOptions
+}
+
+var _ volume.Provisioner = &flockerVolumeProvisioner{}
+
+func (c *flockerVolumeProvisioner) Provision() (*api.PersistentVolume, error) {
```

Comment:
If I am reading it right, you do not support any configuration in StorageClass.Parameters and in PVC.Spec.Selector. In that case, you should check that `c.options.parameters` and `c.options.selector` are empty and throw an error if not.